snhst/reduce_hst_data.py

```diff
@@ -17,178 +17,217 @@
+                   'dolphot_img': {},
+                   'dolphot_sky': {}}
+
+    global topdir
+    topdir = os.getcwd()
 
     # Unzip images
-    os.system('gunzip *.fits.gz')
+    if glob('*.fits.gz'):
+        os.system('gunzip *.fits.gz')
 
     # Download the reference images
-    download_reference_images()
+    if not os.path.exists('ref/'):
+        download_reference_images()
 
     # Copy the raw files into the raw directory
-    copy_raw_data()
+    if not os.path.exists('raw/'):
+        copy_raw_data()
+
+    # output coords from header of ground reference, if not given
+    if center is None and ground_reference is not None:
+        center['ra'], center['dec'] = get_target_coordinates(ground_reference)
+
+    raw_images = get_raw_image_filenames()
 
     # Make sure the position of interest is in the raw frame
     if center is not None:
-        remove_images_without_object(center['ra'], center['dec'])
+        remove_images_without_object(center['ra'], center['dec'], raw_images)
+        options['ra'], options['dec'] = center['ra'], center['dec']
 
     # sort the data into instrument/visit/filter
-    visit_meta_data = sort_raw_data()
+    visit_meta_data = sort_raw_data(raw_images)
+    hst_reference = get_default_reference_hst(visit_meta_data)
+    instrument_string = go_to_visit(hst_reference)
 
     # Make the overall HST template image
-    if reference_image is not None:
-        options['ra'], options['dec'] = get_center_of_image(reference_image)
+    if ground_reference is not None:
+
+        drizzled_template_filename, drizzled_template_catalog = match_and_drizzle('match_template', instrument_string, options)
+        drizzled_template_header = fits.getheader(drizzled_template_filename, extname='SCI')
+
+        # Get the science hdu of the ground reference (may be compressed)
+        reference_hdulist = fits.open(os.path.join(topdir, ground_reference))
+        for reference_hdu in reference_hdulist:
+            if reference_hdu.header.get('NAXIS'):
+                break
+
+        if not os.path.exists('ground_reference.cat'):
+            # Reproject ground image onto drizzled HST frame
+            data, footprint = reproject.reproject_interp(reference_hdu, drizzled_template_header)
+            # Run sep on the ground image to make the ground catalog
+            make_sep_catalog(data, options['sep'], 'match_template/ground_reference.cat')
+
+        # Calculate the offset between the frames
+        # (both catalogs are projected to the WCS of the drizzled template)
+        offsets = wcs.calculate_wcs_offset(drizzled_template_header, drizzled_template_catalog, 'match_template/ground_reference.cat', max_offset=2.)
+    else:
+        offsets = None
+
+    hst_reference_filename, hst_reference_catalog = match_and_drizzle('hst_template', instrument_string, options, offsets)
+
+    # drizzle each visit separately
+    for visit in visit_meta_data:
+        if (visit == hst_reference).all():
```

Comment:
Done, and I also had to rename the directory structure (match_template and hst_template) so that it wouldn't overwrite files from the initial match to the ground reference later in the process.